virttest/utils_time.py

```diff
@@ -0,0 +1,243 @@
+    """
+    Sync guest's timezone to host's.
+
+    This function gets guest's timezone code and compares it with host's
+    timezone code, if they are same, return True, or updates the guest's
+    timezone code to host's timezone code, then return True. If something
+    in this process, return False.
+
+    :param os_type: os type of given guest (eg. linux or windows)
+    :param vm: Virtual machine for vm.
+    :param session: VM session
+    :param zonecode: Host's timezone in code mode.
+    :param zonecity: Host's timezone in city mode.
+
+    :return: True if guest has same timezone with host.
+    """
+    status = -1
+    if os_type == "linux":
+        check_tz_cmd = 'timedatectl | grep "Time zone"'
+        set_tz_cmd = "timedatectl set-timezone %s"
+        list_tz_cmd = "timedatectl list-timezones"
+    else:
+        check_tz_cmd = "tzutil /g"
+        set_tz_cmd = 'tzutil /s "%s"'
+        list_tz_cmd = "tzutil /l"
+    o_guest = session.cmd_output(check_tz_cmd)
+    logging.info("check timezone in guest: %s" % o_guest)
+    if os_type == "linux":
+        guest_tz = re.match(r'^(?:\s+Time zone:\s)(\w+\/\w+|UTC)(?:\s\(\w+'
+                            r',\s)([+|-]\d{4})\)$', o_guest, re.M | re.I)
+        if guest_tz.groups()[1] == zonecode:
+            return (True, session)
+        else:
+            status = session.cmd_status(set_tz_cmd % zonecity)
+    elif os_type == "windows":
+        guest_tzs = []
+        guest_tz = []
+        tz_list = session.cmd_output(list_tz_cmd)
+        tz_list = tz_list.splitlines()
+        for line in tz_list:
+            if not line:
+                continue
+            og = re.search(r'(?:\(UTC([+|-]\d{2}:\d{2})?)', line,
+                           re.M | re.I)
+            if og and og.group(1):
+                og_ = re.sub(r'(\d{2}):(\d{2})', r'\1\2', og.group(1))
+                guest_tz.append(og_)
+                continue
+            elif og:
+                guest_tz.append("+0000")
+                continue
+            og_value = re.search(r'(\S+(?:\s\S+)*)', line, re.M | re.I)
+            if og_value:
+                guest_tz.append(og_value.group(0))
+            guest_tzs.append(guest_tz)
+            guest_tz = []
+
+        for value in guest_tzs:
+            if o_guest == value[1] and value[0] == zonecode:
+                return (True, session)
+
+        # Windows guest's timezone not match host's timezone
+        # get the timezone value for windows guest
+        tz_value = ""
+        for value in guest_tzs:
+            if zonecode == value[0]:
+                tz_value = value[1]
+                break
+        status = session.cmd_status(set_tz_cmd % tz_value)
+
+    if not status:
+        session = vm.reboot(session=session, timeout=600)
+        return (True, session)
+    else:
+        logging.warn("Fail to update timezone in guest")
+        return (False, session)
+
+
+def set_guest_clocksource(session, list_guest, expected):
```

Comment:
`set_guest_clocksource`, `get_guest_clocksource`, `set_host_clocksource` and `get_host_clocksource` can be clubbed together as,
```python
def clock_source_info(list_info, expected=None, session=None, set=False):
    # For guest
    if session:
    # For host
    else:
```